pkg/webhooks/customresourcedefinitions/customresourcedefinitions.go

```diff
@@ -0,0 +1,202 @@
+package customresourcedefinitions
+
+import (
+	"fmt"
+	"net/http"
+	"regexp"
+	"slices"
+	"strings"
+
+	"github.com/openshift/managed-cluster-validating-webhooks/pkg/webhooks/utils"
+
+	admissionregv1 "k8s.io/api/admissionregistration/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	admissionctl "sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+const (
+	WebhookName                    string = "customresourcedefinitions-validation"
+	docString                      string = `Managed OpenShift Customers may not change CustomResourceDefinitions managed by Red Hat.`
+	privilegedServiceAccountGroups string = `^system:serviceaccounts:(kube.*|openshift.*|default|redhat.*|osde2e-[a-z0-9]{5})`
+)
+
+var (
+	timeout                          int32 = 2
+	allowedUsers                           = []string{"system:admin", "backplane-cluster-admin"}
+	sreAdminGroups                         = []string{"system:serviceaccounts:openshift-backplane-srep"}
+	privilegedServiceAccountGroupsRe       = regexp.MustCompile(privilegedServiceAccountGroups)
```

Comment:
Could we use `utils.PrivilegedServiceAccountGroups` instead there was a recent finding that needed to update this regex string?

Reply:
added in a973e244c24dcccbde17fd4d3078e746d410ecb6